packages/sil-governance/src/routes/dashboard/components/ProposalsList.tsx

```diff
@@ -34,16 +44,78 @@ const getFilter = (filterType: ElectionFilter, currentUser: Address | null): Pro
   }
 };
 
-function compareById<E extends { readonly id: number }>(element1: E, element2: E): number {
-  return element1.id - element2.id;
+enum ComparatorId {
+  IdDescending,
+  ExpiryDate,
+  StartDate,
+  Vote,
+}
+
+type ProposalsComparator = (proposal1: ProposalProps, proposal2: ProposalProps) => number;
+
+const compareByIdDescending: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.id < proposal2.id ? 1 : -1;
+};
+const compareByExpiryDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.expiryDate.getTime() - proposal2.expiryDate.getTime();
+};
+const compareByStartDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.startDate.getTime() - proposal2.startDate.getTime();
+};
+const compareByVote: ProposalsComparator = (proposal1, proposal2): number => {
+  if (proposal1.vote === undefined && proposal2.vote === undefined) return 0;
+  if (proposal1.vote === undefined) return 1;
+  if (proposal2.vote === undefined) return -1;
+  return proposal1.vote - proposal2.vote;
+};
+
+interface Props {
+  readonly filterType: ElectionFilter;
 }
 
 const ProposalsList = ({ filterType }: Props): JSX.Element => {
-  const proposals = ReactRedux.useSelector((state: RootState) => state.proposals.proposals);
-  const governor = ReactRedux.useSelector((state: RootState) => state.extension.governor);
-  const blockchain = ReactRedux.useSelector((state: RootState) => state.blockchain);
+  const proposals = useSelector((state: RootState) => state.proposals.proposals);
+  const governor = useSelector((state: RootState) => state.extension.governor);
+  const blockchain = useSelector((state: RootState) => state.blockchain);
+
+  const [comparatorId, setComparatorId] = useState<ComparatorId>(ComparatorId.IdDescending);
+
+  const changeComparator = (selectedItem: SelectFieldItem): void => {
+    switch (selectedItem.name) {
+      case ComparatorLabels.ExpiryDate:
+        setComparatorId(ComparatorId.ExpiryDate);
+        break;
+      case ComparatorLabels.StartDate:
+        setComparatorId(ComparatorId.StartDate);
+        break;
+      case ComparatorLabels.Vote:
+        setComparatorId(ComparatorId.Vote);
+        break;
+      default:
+        setComparatorId(ComparatorId.IdDescending);
+    }
+  };
+
+  const { form } = useForm({ onSubmit: () => {} });
 
   const filter = getFilter(filterType, governor ? governor.address : null);
+
+  let comparator;
```

Comment:
Can you make this typed: `let comparator: ProposalsComparator`? Thiy may require throwing an exception in the switch default to let TypeScript understand it will always be set.